### PR TITLE
e2e visual testing - Fix dynamic article content and top-above-nav 

### PR DESCRIPTION
**Pull request by arelra**
## What does this change?

Foundational work for Commercial e2e visual testing, primarily around deterministic runs:

- separate snapshot tests from regular e2e tests
- stub API responses for dynamic content (most viewed, headlines etc)
- a workaround for top-above-nav intermittently rendering an MPU which seems to be caused by the interaction between GAM and Cypress

TODO:
- fix the 'right' slot as it can be an MPU or DMPU
- fix the merchandising slots as the creative text can differ from run to run
- investigate: headlines section even though the api response is stubbed seems to randomise the order

## Does this change need to be reproduced in dotcom-rendering ?

- [x] No
- [ ] Yes (please indicate your plans for DCR Implementation)

## Screenshots

|     |  Before | After  |
|-----|---|---|
| top-above-nav |  Screenshot 2022-10-10 at 17 15 57 |  Screenshot 2022-10-10 at 17 17 50 |
|  dynamic content   |  Screenshot 2022-10-10 at 17 19 31 |  Screenshot 2022-10-10 at 17 19 57 |

### Tested

- [x] Locally
- [ ] On CODE (optional)

### CODE REVIEW

File: cypress/snapshot/snapshot-standard-article.cy.ts [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+				req.url = rewriteUrl(req.url);
+			}
+		},
+	);
+
+/**
+ * GAM intermitently renders a 320x250 MPU ad in the top-above-nav slot when in a Cypress test
+ * For now we hardcode size targeting on the GAM request while we investigate
+ */
+const rewriteGamRequestForTopAboveNav = (reqUrl: string) => {
```
Comment (Jakeii) [inline cypress/snapshot/snapshot-standard-article.cy.ts:45]: Nice workaround!

**Review by Jakeii — APPROVED** [review]

File: cypress/snapshot/snapshot-standard-article.cy.ts [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+	console.log('URL before rewrite:', reqUrl);
```
Comment (SiAdcock) [inline cypress/snapshot/snapshot-standard-article.cy.ts:46]: Do we wanna keep the `console.log`s?

Comment (arelra) [reply cypress/snapshot/snapshot-standard-article.cy.ts:46]: Yes sorry I should have mentioned. I'm still investigating why googletag gets the ad size wrong so would like to keep these in for now. It only logs to the window in the Cypress test so it doesn't affect anything.

**Review by SiAdcock — APPROVED** [review]
This is awesome / exciting / beautiful  😄 💎

**Pull request merged**